controllers/Users.php

```diff
@@ -221,6 +221,22 @@ public function preview_onImpersonateUser($recordId)
         Flash::success(Lang::get('rainlab.user::lang.users.impersonate_success'));
     }
 
+    /**
+     * Unsuspend this user
+     */
+    public function preview_onUnsuspendUser($recordId)
+    {
+        $model = $this->formFindModelObject($recordId);
+
+        $model->unsuspend();
+
+        Flash::success(Lang::get('rainlab.user::lang.users.unsuspend_success'));
+
+        if ($redirect = $this->makeRedirect('update-close', $model)) {
+            return $redirect;
+        }
```

Comment:
What's the point of this code here? What does it do under what different conditions?

Reply:
As it redirects back to the same page, it removes the button. As the user is no longer suspended. That way the user gets some additional feedback that the user is indeed unsuspended (as well as having the flash message).

Reply:
@ashleighsims Why have that conditional at all then? Seems like it would be better to just do `return Redirect::refresh();`

Reply:
@LukeTowers I was probably just following the pattern that the other controller methods have with regards to the redirect. With the functionality being similar to the unban and activate. It can be changed if that is preferred?

Reply:
I think it would be preferred, it's more explicit about what exactly is going on / why.